Weekly_Work/Week2/Code_Changes_Git.py

```diff
@@ -0,0 +1,159 @@
+import json
+import datetime
+from Commit import Commit
+import pandas as pd
+import matplotlib.pyplot as plt
+import utils
+
+class Code_Changes_Git(Commit):
+    """Class for Code_Changes for Git repositories.
+    
+    Objects are instantiated by specifying a file with the
+    commits obtained by Perceval from a set of repositories.
+        
+    :param path: Path to file with one Perceval JSON document per line
+    """
+    
+    def __init__(self, path, date_range=(None, None), source_code_exclude_list=None):
+        """
+        Initilizes self.df, the dataframe with one commit per row.
+        The source_code_exclude_list parameter is a list which contains file extensions
+        which are to be ignored. All possible file extensions are allowed. 
+        For files without a standard ".xyz" extension, like LICENCE or AUTHORS, the "others" 
+        extension is used. 
+        
+        :param path: Path to file with one Perceval JSON document per line
+        :param date_range: A tuple which represents the start and end date of interest
+        :param source_code_exclude_list: Files extensions to exclude. For example, 
+            source_code_exclude_list = ["py", "other", "gitignore"]
+        """
+        
+        super().__init__(path, date_range, source_code_exclude_list)
+
+        
+    def total_count(self):
+        """
+        Get a naive count of the number of commits in the Perceval data.
+        Note that some commits may be repeated and so total_count may 
+        overshoot.
+        """
+        return len(self.df.index)
+    
+    def compute(self, empty=True, merge=True, master_only=False):
+        """Count number of commits of different types
+        
+        :param empty: Include empty commits
+        :param merge: Include merge commits
+        :param master_only: Include only commits made on master branch
+        """
+        count_per_repo = dict()
+        for repo_url in self.repo_urls:
+            df = self.df[self.df['repo'] == repo_url]
+            
+            if master_only:
+                count_per_repo[repo_url] = self._count_master_only(repo_url, empty)
+
+            else:
+                if not empty:
+                    df = df[df['files_action'] != 0]
+                if not merge:
+                    df = df[df['merge'] == False]
+                count_per_repo[repo_url] = df['hash'].nunique()
+            
+        return count_per_repo
+    
+    def compute_timeseries(self, period="month", plot_chart=False):
```

Comment:
If we want to do this (which I'm still not completely sure, but I don't oppose for now), i think we could do it in the parent class, couldn't we?

Reply:
If you are talking about `count_per_repo`, I alreadly changed the code so that the metric is calculated as a whole for the file, irrespective of how many repositories are present in it, just as you suggested above.